a/apache-arrow/arrow_ubi_8.7.sh

```diff
@@ -20,8 +20,8 @@
 
 PACKAGE_NAME=arrow
 PACKAGE_URL=https://github.com/apache/arrow.git
-PACKAGE_VERSION=${1:-go/v10.0.1}
-GO_VERSION=${GO_VERSION:-1.19.6}
+PACKAGE_VERSION=${1:-go/v12.0.1}
+GO_VERSION=${GO_VERSION:-1.20.3}
```

Comment:
Updating go version every time is creating maintenance headache. So @vinodk99  please see if we can make this dynamic or any additional step to be added so we wont need to update build script for go version only. I'm ok for this PR but just question to address .